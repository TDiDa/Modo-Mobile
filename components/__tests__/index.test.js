const OLD_NODE_ENV = process.env.NODE_ENV;
process.env.NODE_ENV = 'development';
const warnSpy = jest.spyOn(console, 'warn').mockImplementation(() => {});
const modo = require('..');

describe('modo', () => {
  afterAll(() => {
    process.env.NODE_ENV = OLD_NODE_ENV;
  });

  it('exports modules correctly', () => {
    expect(Object.keys(modo)).toMatchSnapshot();
  });

  it('should hint when import all components in dev mode', () => {
    expect(warnSpy).toBeCalledWith(
      'You are using a whole package of modo-mobile, please use https://www.npmjs.com/package/babel-plugin-import to reduce app bundle size.',
    );
    warnSpy.mockRestore();
  });
});
