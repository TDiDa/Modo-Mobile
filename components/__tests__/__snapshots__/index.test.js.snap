// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modo exports modules correctly 1`] = `
Array [
  "Button",
  "Icon",
  "NoticeBar",
  "Agree",
  "List",
  "Stepper",
  "Popup",
  "NumberKeyboard",
  "ActivityIndicator",
  "version",
]
`;
